scripts/march-to-cpu-opt

```diff
@@ -121,31 +149,61 @@ def get_vlen(ext_dict):
             vlen = max(vlen, zvelen)
     return vlen
 
-def conver_arch_to_qemu_cpu_opt(march):
-    if len(march) < 5:
-        return None
+def get_elen(ext_dict, xlen):
+    elen = xlen
 
-    ext_dict = parse_march(march)
+    if "zve32x" in ext_dict or "zve32f" in ext_dict:
+        elen = 32
+    if "zve64x" in ext_dict or "zve64f" in ext_dict or "zve64d" in ext_dict:
+        elen = 64
 
-    cpu_opt = []
-    cpu_opt.append(march[0:4]) # rv32 or rv64
+    return elen
 
-    vlen = get_vlen(ext_dict)
+def print_qemu_cpu():
+    cpu_options = []
+    cpu_options.append("rv{0}".format(CPU_OPTIONS['xlen']))
 
-    if vlen != 0:
-        cpu_opt.append("vlen=%d" % vlen)
+    if CPU_OPTIONS['vlen']:
+        cpu_options.append("vlen={0}".format(CPU_OPTIONS['vlen']))
+        # Enable fill one semantic for tail/mask agnostic, this could discover
+        # more potential bug.
+        cpu_options.append("rvv_ta_all_1s=true")
+        cpu_options.append("rvv_ma_all_1s=true")
 
     disable_all_fd = False
-    for ext in ext_dict.keys():
-        if ext in EXT_OPTS:
-            cpu_opt.append(EXT_OPTS[ext])
+    for ext in CPU_OPTIONS['extensions']:
+        if ext in QEMU_EXT_OPTS:
+            cpu_options.append(QEMU_EXT_OPTS[ext])
+
         if ext in ['zhinx', 'zfinx', 'zdinx']:
             disable_all_fd = True
+
     if disable_all_fd:
-        cpu_opt.append("f=false")
-        cpu_opt.append("d=false")
-    return ",".join(cpu_opt)
+        cpu_options.append("f=false")
+        cpu_options.append("d=false")
+
+    return ",".join(cpu_options)
+
+def print_spike_isa():
+    cpu_options = []
+    cpu_options.append("rv{0}".format(CPU_OPTIONS['xlen']))
```

Comment:
If the string is initialized here, f"rv{CPU_OPTIONS['xlen']}" might be easier to read

Reply:
Discussed and while I agree it'd be better to fix upstream and pull it in.